routers/repo/issue_dependency_add.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+)
+
+// AddDependency adds new dependencies
+func AddDependency(c *context.Context) {
+
+	// TODO: should should an issue only have dependencies in it's own repo?
+
+	depID, err := strconv.ParseInt(c.Req.PostForm.Get("newDependency"), 10, 64)
+	if err != nil {
+		c.Handle(http.StatusBadRequest, "issue ID is not int", err)
+		return
+	}
+
+	issueIndex := c.ParamsInt64("index")
+	issue, err := models.GetIssueByIndex(c.Repo.Repository.ID, issueIndex)
+	if err != nil {
```

Comment:
Please minimize that as shown above

Reply:
Don't think it is worth it, I would need to declare both `issue` and `err` before calling the function which wouldn't result in better readability.